src/modules/html/producer/html_producer.cpp

```diff
@@ -302,7 +369,10 @@ class html_client
         {
             std::lock_guard<std::mutex> lock(frames_mutex_);
 
-            frames_.push(std::make_pair(now(), core::draw_frame(std::move(frame))));
+            core::draw_frame new_frame = core::draw_frame(std::move(frame));
+            last_generated_frame_      = new_frame;
```

Comment:
@Julusian Could you talk me through how the move semantics works here? As far as I can tell, the draw_frame assignment operator moves the content of new_frame to last_generated_frame_ which would leave new_frame invalid.

Reply:
> As far as I can tell, the draw_frame assignment operator moves the content of new_frame to last_generated_frame_ which would leave new_frame invalid.

The parameter of the assignment operator is not a reference, so will be a copy of `new_frame` making it safe to mutate. You have made me question this now, but this same pattern has been in place for `draw_frame` for years, with the `core::frame_producer` doing it with any `draw_frame` that gets produced by every producer.

Reply:
Ok, I see it now. I missed the implicit call to the copy constructor during the assignment. It looks OK.